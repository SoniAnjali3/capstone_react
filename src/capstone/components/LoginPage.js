import React from 'react';
import { withFormik, Field, Form } from 'formik';
import * as Yup from 'yup';
import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import '../../App.css';

const LoginForm = ({ errors, touched, isSubmitting }) => (
  <div>
    <h1>Login</h1>
    <Form>
      <div className='form-group'>
        <Field
          type='text'
          name='EmailID'
          placeholder='EmailID'
          className='form-control'
        />
        {touched.EmailID && errors.EmailID && (
          <div style={{ color: 'red' }}>{errors.EmailID}</div>
        )}
      </div>
      <div className='form-group'>
        <Field
          type='text'
          name='Password'
          placeholder='Password'
          className='form-control'
        />
        {touched.Password && errors.Password && (
          <div style={{ color: 'red' }}>{errors.Password}</div>
        )}
      </div>
      <br></br>
      <button type='submit' disabled={isSubmitting} className='btn btn-primary'>
        Submit
      </button>
    </Form>
  </div>
);

const FormikLoginForm = withFormik({
  mapPropsToValues({ EmailID, Password }) {
    //initial value passed from App.js will be set in these parameters
    return {
      EmailID: 'anjali@gmail.com' || '',
      Password: 'anjali' || '',
    };
  },
  validationSchema: Yup.object().shape({
    EmailID: Yup.string()
      .email('Email is not valid')
      .required('EmailID is required'),
    Password: Yup.string().required('Password is required'),
  }),
  handleSubmit(values, { props, resetForm, setSubmitting, setErrors }) {
    //console.log(values);
    setTimeout(() => {
      resetForm();
      // alert(JSON.stringify(values));
      // console.log('values : ' + values);
      props.onSave(values);
      setSubmitting(false);
    }, 2000);
  },
})(LoginForm);

export default FormikLoginForm;
// export default class LoginForm  extends React.Component {
//     constructor(props) {
//         super(props);

//         this.onSubmit = this.onSubmit.bind(this);
//     }

//     onSubmit(event) {
//         event.preventDefault();
//         var user = {};
//         user.EmailID = this.refs.EmailID.value;
//         user.Password = this.refs.Password.value;
//         this.props.onSave(user);
//     }

//     render() {
//         return(
//             <form>
//                 <h1>Login</h1>
//                 EmailID: &nbsp;
//                 <input type="text" ref="EmailID"></input><br/><br/>
//                 Password: &nbsp;
//                 <input type="text" ref="Password"></input><br/><br/>
//                 <button type="submit" value="Save" onClick={this.onSubmit} />
//             </form>
//         );
//     }
// }
