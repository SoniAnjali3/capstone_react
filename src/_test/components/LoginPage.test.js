import React from 'react';
import { mount, shallow } from 'enzyme';
import LoginPage from '../../capstone/components/LoginPage';

describe('Test LoginPage using Shallow rendering', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LoginPage />);
  });

  it('has 2 text input elements', () => {
    expect(wrapper.find('input').length).toEqual(2);
  });

  it('has a single button', () => {
    expect(wrapper.find('button').length).toEqual(1);
  });

  it('has a single h1', () => {
    expect(wrapper.find('h1').length).toEqual(1);
  });

  it('has a render h1 correctly', () => {
    expect(wrapper.find('h1').render().text()).toEqual('Login');
  });
});

describe('Test LoginPage using Full DOM rendering', () => {
  let mountWrapper;

  beforeEach(() => {
    mountWrapper = mount(<LoginPage />);
  });

  afterEach(() => {
    mountWrapper.unmount();
  });

  it('has 2 text input elements', () => {
    expect(mountWrapper.find('input').length).toEqual(2);
  });

  it('has a single button', () => {
    expect(mountWrapper.find('button').length).toEqual(1);
  });
});

describe('LoginPageSnapShot', () => {
  let wrapperFullDOM;
  beforeEach(() => {
    wrapperFullDOM = mount(<LoginPage></LoginPage>);
  });
  test('LoginPage renders correctly', () => {
    expect(wrapperFullDOM).toMatchSnapshot(); //first time takes copy of snapshot and afterwards compare with old snapshot
  });
});
