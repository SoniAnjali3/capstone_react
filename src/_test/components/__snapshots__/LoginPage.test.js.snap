// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPageSnapShot LoginPage renders correctly 1`] = `
<LoginForm>
  <form>
    <h1>
      Login
    </h1>
    EmailID:  
    <input
      type="text"
    />
    <br />
    <br />
    Password:  
    <input
      type="text"
    />
    <br />
    <br />
    <button
      onClick={[Function]}
      type="submit"
      value="Save"
    />
  </form>
</LoginForm>
`;
